src/commands/importFromRegistry.ts

```diff
@@ -0,0 +1,168 @@
+          placeHolder: "username",
+          prompt: `Username for server '${serverName}'`,
+          validateInput: ((value) => {
+              return value.length > 0 ? "" : "Mandatory field";
+          }),
+        }, cancellationToken);
+        if (cancellationToken.isCancellationRequested || username === undefined) {
+          return;
+        }
+        if (serversMissingUsernames.length > 1) {
+          const items = [
+            `Use username '${username}' for all servers`,
+            `Enter usernames individually for ${serversMissingUsernames.length - 1} more server(s)`,
+            `Cancel`].map((label) => {
+              return { label };
+            });
+          const result = await vscode.window.showQuickPick(items, {
+            canPickMany: false,
+            ignoreFocusOut: true,
+          }, cancellationToken);
+          if (cancellationToken.isCancellationRequested || result === undefined || result.label === items[2].label) {
+            return;
+          } else if (result.label === items[0].label) {
+            for (serverName of serversMissingUsernames) {
+              serverDefinitions[serverName].username = username;
+            }
+          } else {
+            for (serverName of serversMissingUsernames) {
+              username = await vscode.window.showInputBox({
+                ignoreFocusOut: true,
+                prompt: `Username for server '${serverName}'`,
+                validateInput: ((value) => {
+                    return value.length > 0 ? "" : "Mandatory field";
+                }),
+                value: username,
+              }, cancellationToken);
+              if (cancellationToken.isCancellationRequested || username === undefined) {
+                return;
+              }
+              serverDefinitions[serverName].username = username;
+            }
+          }
+        }
+      }
+
+      let reusePassword;
+      let password;
+      for (const serverName of newServerNames) {
+        if (!reusePassword) {
+          password = await vscode.window.showInputBox({
+            ignoreFocusOut: true,
+            password: true,
+            placeHolder: "Password to store in keychain",
+            prompt: `For connection to InterSystems server '${serverName}'
+              as ${serverDefinitions[serverName].username}`,
+            validateInput: ((value) => {
+                return value.length > 0 ? "" : "Mandatory field";
+            }),
+          }, cancellationToken);
+
+          if (!password) {
+            return;
+          }
+
+          if (cancellationToken.isCancellationRequested) {
+            return;
+          }
+        }
+
+        if (reusePassword === undefined) {
+          const items = [
+            `Use this password for all servers`,
+            `Enter passwords individually for ${newServerNames.length - 1} more server(s)`,
+            `Cancel`].map((label) => {
+              return { label };
+            });
```

Comment:
![image](https://user-images.githubusercontent.com/6726799/101466705-5a8c5f00-3939-11eb-9027-45e002fd858c.png)

I know mine was an edge case, having only one server in the registry, but I think it should still be handled better.

When it **is** appropriate to ask this question I suggest adding `placeHolder` text such as "Store the same password for remaining N server(s)?" and make the options "No" (the default), "Yes", and "Cancel import"